lib/datadog/tracing/contrib/rails/patcher.rb

```diff
@@ -5,11 +5,59 @@
 require_relative 'middlewares'
 require_relative 'utils'
 require_relative '../semantic_logger/patcher'
+require_relative 'ext'
 
 module Datadog
   module Tracing
     module Contrib
       module Rails
+        # Patcher to begin span on Rails routing
+        module RoutingRouteSetPatch
+          def call(*args, **kwargs)
+            result = nil
+
+            configuration = Datadog.configuration.tracing[:rails]
+
+            Tracing.trace(
+              Ext::SPAN_ROUTE,
+              service: configuration[:service_name],
+              span_type: Tracing::Metadata::Ext::HTTP::TYPE_INBOUND,
+            ) do |span|
+              span.set_tag(Tracing::Metadata::Ext::TAG_COMPONENT, Ext::TAG_COMPONENT)
+              span.set_tag(Tracing::Metadata::Ext::TAG_OPERATION, Ext::TAG_OPERATION_ROUTING)
+              result = super
+            end
+
+            result
+          end
+        end
+
+        # Patcher to trace rails routing done by JourneyRouter
+        module JourneyRouterPatch
+          def find_routes(*args, **kwargs)
+            result = super
+
+            if Datadog::Tracing.enabled? && (span = Datadog::Tracing.active_span)
+              if result.any?
+                datadog_route = result.first[2].path.spec.to_s
+              end
```

Comment:
Otherwise `datadog_route` is `nil`, but we still record it as the router failing to match.